sdk/src/wallet/core/mod.rs

```diff
@@ -353,55 +374,85 @@ where
                 }
 
                 // If ids are provided, only return them and no other outputs.
-                if filter.account_ids.is_none() && filter.foundry_ids.is_none() && filter.nft_ids.is_none() {
+                if filter.account_ids.is_none()
+                    && filter.anchor_ids.is_none()
+                    && filter.foundry_ids.is_none()
+                    && filter.nft_ids.is_none()
+                    && filter.delegation_ids.is_none()
+                {
                     filtered_outputs.push(output.clone());
                 }
```

Comment:
A bit weird this this `if` tbh. Also ... seems like something that can easily introduce a bug. I wonder if there's a better way. Also ... is the comment correct?

Reply:
I'll put your comment but without the always as it's a bit too strong, it still depends on the other filtering options